src/main/java/uk/ac/ebi/eva/utils/VcfNumberOfLinesEstimator.java

```diff
@@ -0,0 +1,137 @@
+ * estimated total number of lines similar to the real one.
+ * <p>
+ * In case of small VCF the NUMBER_OF_LINES will be 0 and no % will be printed in {@link StepProgressListener#afterChunk}
+ */
+public class VcfNumberOfLinesEstimator {
+    private static final Logger logger = LoggerFactory.getLogger(VcfNumberOfLinesEstimator.class);
+
+    private static final int NUMBER_OF_LINES = 100;
+
+    public int estimateVcfNumberOfLines(String vcfFilePath) {
+        logger.debug("Estimating the number of lines in the VCF file {}", vcfFilePath);
+        int estimatedTotalNumberOfLines = 0;
+
+        String vcfHead;
+        String vcfSection;
+
+        try {
+            vcfHead = retrieveVcfHead(vcfFilePath);
+            vcfSection = retrieveVcfSection(vcfFilePath);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading VCF " + vcfFilePath, e);
+        }
+
+        if (!vcfSection.isEmpty()) {
+            File vcfHeadFile;
+            File vcfSectionFile;
+
+            try {
+                vcfHeadFile = FileUtils.newGzipFile(vcfHead, "vcfHeadFile");
+                vcfSectionFile = FileUtils.newGzipFile(vcfSection, "vcfSectionFile");
+            } catch (IOException e) {
+                throw new RuntimeException("Error while creating zip file", e);
+            }
+
+            int vcfFileSize = (int) new File(vcfFilePath).length();
+            int singleVcfLineSize = (int) (vcfSectionFile.length() / NUMBER_OF_LINES);
+            int vcfHeadFileSize = (int) vcfHeadFile.length();
+
+            estimatedTotalNumberOfLines = ((vcfFileSize - vcfHeadFileSize) / singleVcfLineSize);
+        }
+
+        logger.info("Estimated number of lines in VCF file: {}", estimatedTotalNumberOfLines);
+        return estimatedTotalNumberOfLines;
+    }
+
+    /**
+     * @param vcfFilePath location of the VCF to parse
+     * @return the head of the VCF
+     * @throws IOException
+     */
+    private String retrieveVcfHead(String vcfFilePath) throws IOException {
+        String vcfHead = "";
+
+        Scanner scanner = new Scanner(new GZIPInputStream(new FileInputStream(vcfFilePath)));
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            if (line.startsWith("#")) {
+                vcfHead += line + "\n";
+            }else {
+                break;
+            }
+        }
+        scanner.close();
+
+        return vcfHead;
+    }
+
+    /**
+     * @param vcfFilePath location of the VCF to parse
+     * @return first NUMBER_OF_LINES of VCF, empty sting in case of VCF smaller than NUMBER_OF_LINES
+     * @throws IOException
+     */
+    private String retrieveVcfSection(String vcfFilePath) throws IOException {
+        String vcfSection = "";
+
+        int lineCount = NUMBER_OF_LINES;
+        Scanner scanner = new Scanner(new GZIPInputStream(new FileInputStream(vcfFilePath)));
+        while (scanner.hasNextLine()) {
```

Comment:
Making this
```java
while (scanner.hasNextLine() && lineCount > 0) {
```
and removing the break would make a bit more clear to see when does the loop actually stop.